### PR TITLE
feat: docker solution for final-api

### PR DESCRIPTION
**Pull request by ondrej-hosak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
+## How can I use this solution?
+* Clone this repository
+* [Install Docker](https://docs.docker.com/engine/installation/linux/)
+* Clone [Final-API](https://github.com/AVGTechnologies/final-api)
+* Adjust the Final-API configs
+* Prepare tsd validator in your final-api project
+* Prepare env file in your final-api project
+* Copy necessary certificates to ./certs
+* Run the make script to prepare images
+* Run the docker-compose up script to launch the docker for you
```
Comment (petrSchreiber) [inline README.md]: I would recommend to highlight the command as `docker-compose up`

**Comment by panjan** [comment]
This is very similar to a subset of the [existing solution](https://github.com/final-ci/docker-final-base-image/tree/iz_docker_images). Would it be better to adjust what we have rather than starting over? What do you think @ireznice, @xBajci, @petrSchreiber, @ondrej-hosak ?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+redis:
+  image: redis:3.0.3
+  restart: always
+
+api:
+ image: final-ci/final-api:latest
```
Comment (Trawais) [inline docker-compose.yml]: incorrect indent

**Review by Trawais — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,2 +1,22 @@
-# final-ci-docker
-Docker files for final-ci service.
+# Dockerized final-api application
+
+This solution allows running Dockerized version of [Final-API](https://github.com/AVGTechnologies/final-api).
+
+## What does it mean "dockerised"? What is the motivation?
```
Comment (ireznice) [inline README.md]: Hey, is it dockerized or dockerised? you should choose one 'version' and use it everywhere :)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,22 @@
...
+Docker is a solution allowing definition of service containers, which can
+be used for comfortable service deployment and cleanup.
+
+This version of final-api targets to bring more comfort to the
+developers.
+
+## How can I use this solution?
+* Clone this repository
+* [Install Docker](https://docs.docker.com/engine/installation/linux/)
```
Comment (ireznice) [inline README.md:14]: you should tell to the user that you also need docker-compose and its version as well.

File: base-image/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,50 @@
+FROM       ubuntu:14.04.2
+MAINTAINER Automation Development team <eng.op.autodev@avg.com>
```
Comment (ireznice) [inline base-image/Dockerfile]: don't know whether it is a good idea to give this email address on github.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,22 @@
...
+* Clone [Final-API](https://github.com/AVGTechnologies/final-api)
+* Adjust the Final-API configs
+* Prepare tsd validator in your final-api project or in ./examples/final-api/bin
+* Prepare config/.env file in your final-api project or in ./examples/final-api/config
+* Copy necessary certificates to ./base-image/certs (CRT format)
+* Run `make` to prepare images
```
Comment (ireznice) [inline README.md:21]: I have to run sudo make ?

File: README.md [hunk 7]
```diff
@@ -1,2 +1,22 @@
...
+
+## What does it mean "dockerised"? What is the motivation?
+Docker is a solution allowing definition of service containers, which can
+be used for comfortable service deployment and cleanup.
+
+This version of final-api targets to bring more comfort to the
+developers.
+
+## How can I use this solution?
+* Clone this repository
```
Comment (panjan) [inline README.md:13]: When we have more services than just final-api, do I have to clone and mount all of them?

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+redis:
```
Comment (panjan) [inline docker-compose.yml:1]: I am missing postgres here. Do I need to host and configure it on my own?

File: README.md (hunk 7 above)
Comment (ondrej-hosak) [reply README.md:13]: No they will be cloned from master but you still have to provide configuration. On the other hand you can override your sources with mounted volume. Later on we will implement solution, that you can start service from other origins and branches/commits.

File: docker-compose.yml (hunk 8 above)
Comment (ondrej-hosak) [reply docker-compose.yml:1]: As discussed on one of our meeting, we don't need stand-alone version now. So the current solution connects to one of our existing solution.

**Pull request merged**